<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Fall 2022 Lab Entries</title>
    <link rel="stylesheet" href="fall.css">
</head>
<body>
    <div id="lab-entries-container">
    <div class="lab-entry">
        <h3 class="date">November 1, 2022</h3>
        <p ondrop="drop(event)" ondragover="allowDrop(event)" class="content">
            <strong>Writing</strong>
            <br><br>
            I am working on shortening the Introduction section of the ISA writeup for the purpose of this Abstract.
            <br><br>
            <u>Hereâ€™s the original text.</u>
            <br><br>
            <i>
            Cancer is the result of an evolutionary process [cite Nowell] and there exists a wealth of genomic data through which we
            can better understand cancer through this evolutionary lens. More precisely, methods have been developed that have the
            capacity to infer the evolutionary history of a tumor in the form of tumor phylogenies. Through the inference of tumor
            histories, we can move the needle ever more slightly towards a greater understanding of cancer. One specific use case
            for these tumor phylogenies may be to offer insights that will guide prognosis. Furthermore, in understanding the
            acquisition and losses of mutations throughout a tumor's evolutionary history, more effective treatments may be offered.
            For example, a commonly used approach to treat cancer is combinatorial drug therapy. While cominbatorial drug therapy is
            oftentimes more effective than the usage of a single drug in many cases, it is within the interest of patients' safety
            to minimize the number of drugs used as each drug carries with it its own side effects and toxicities. Tumor phylogenies
            can help clinicians make specific decisions that can most effectly treat the cancer while simultaneously minimizing
            toxicity.
            <br><br>
            In order to infer tumor phylogenies from sequencing data, many assumptions needed to be made about the underlying cancer
            biology in order to make computation more feasible. One dominant assumption begins with the view of tumor evolution as a
            monoclonal process, i.e. originating from a single cell. This assumption makes sense in the face of cancer biology as it
            is suspected that the majority of cancers have monoclonal origins. As such, this assumption of monoclonal evolution has
            largely remained intact throughout the history of the field of reconstructing tumor evolutionary histories. However, not
            all assumptions experienced the same type of luxury. The Infinite Sites Assumption (ISA), states two primary assumptions
            about the nature of tumor phylogenies. (1) no mutation occurs more than once (i.e. no homoplasy) and (2) once a mutation
            is gained, it is never lost (i.e. no back mutations). Many methods have been developed that infer the evolutionary
            history of a tumor as it conforms to the ISA [cite papers]. In accordance, methods have been designed specifically to
            compare these tumor phylogenies [cite papers].
            <br><br>
            It is known that these assumptions are imperfect. However, in light of research showing the role of mutational losses in
            a tumor's developing drug resistance, it is increasingly more important to move away from the ISA. Especially with
            regards to breast and ovarian cancers, a back mutation of the BRCA2 gene may lead to resistance towards the commonly
            used chemotherapy drug cisplatin [cite papers].
            <br><br>
            The k-Dollo model of tumor evolution relaxes the constraint of no mutational losses. The trees constructed under this
            model better explain the true underlying biology of a tumor, but the pace of distance measures research has fallen short
            of the advances within tumor phylogeny inference. Specifically, no existing distance measures have been designed for
            trees constructed under the k-Dollo model. While these existing distance measures may not necessarily break when given
            k-Dollo phylogenies as input, the output generated will likely yield poor distance measures.
            <br><br>
            This paper contributes a deeper analysis as to why existing distance measures that were designed to work with ISA trees
            cannot be applied to k-Dollo phylogenies as is. After addressing demonstrating this problem, we propose a framework
            formalized as the Generalized Matching Distance (GMD) Problem that can will resolve the problem of applying
            ISA-dependent methods on non-ISA tumor trees.
            </i>
            <br><br>
            <u>Here is the revision copy.</u>
            <br><br>
            <i>
            Cancer is the result of an evolutionary process [cite Nowell]. Methods are being developed that can infer this process
            from sequencing data in the form of tumor phylogenies. Tumor phylogenies can be applied in the prognosis of and as a
            tool to guide cancer treatment plans. A common assumption used in the process of inferring tumor phylogenies is the
            Infinite Sites Assumption (ISA). The ISA, states two primary assumptions about the nature of tumor phylogenies. (1) no
            mutation occurs more than once (i.e. no homoplasy) and (2) once a mutation is gained, it is never lost (i.e. no back
            mutations). Many methods have been developed that infer the evolutionary history of a tumor as it conforms to the ISA
            [cite papers]. In accordance, methods have been designed specifically to compare these tumor phylogenies [cite papers].
            In light of research showing the role of mutational losses in a tumor's developing drug resistance both via copy number
            aberrations and true back mutations, it is increasingly more important to move away from the ISA. One such proposed
            model that relaxes the ISA is the k-Dollo model that allows for losses of mutations. While many inference methods have
            adopted this new model of tumor phylogenies, existing distance measures are not adequate at comparing them. Our research
            contributes a deeper analysis as to why existing distance measures that were designed to work with ISA trees cannot be
            applied to k-Dollo phylogenies. After demonstrating this problem, we propose a framework formalized as the Generalized
            Matching Distance (GMD) Problem that can resolve the problem of applying ISA-dependent methods on non-ISA tumor trees.
            </i>
            <br><br>
            <strong>End writing</strong>
        </p>

    </div>
    <div class="lab-entry">
        <h3 class="date">November 2, 2022</h3>
        <p ondrop="drop(event)" ondragover="allowDrop(event)" class="content">Accumulating all figures. (Updated November 17, 2022)</p>
        <strong style="display: block;">4 mutations, Variable losses</strong>
        <img src="./images/4mut_varying_losses_DEPTH_CASET.png" alt="" srcset="">
        <img src="./images/4mutations_VariableLosses_PARENT_CASET.png" alt="" srcset="">
        <img src="./images/4mut_varying_losses_lineage_CASET.png" alt="" srcset="">
        <img src="./images/4mutations_VariableLosses_DEPTH_DISC.png" alt="" srcset="">
        <img src="./images/4mutations_VariableLosses_PARENT_DISC.png" alt="" srcset="">
        <img src="./images/4mutations_VariableLosses_LINEAGE_DISC.png" alt="" srcset="">
        <img src="./images/4mutations_VariableLosses_LINEAGE_PARENT-CHILD.png" alt="">
        <img src="./images/4mutations_VariableLosses_PARENT_PARENT-CHILD.png" alt="">
        <br><br>
        <strong style="display: block;">5 mutations, Variable losses</strong>
        <img src="./images/5mutations_VariableLosses_LINEAGE_CASET.png" alt="" srcset="">
        <br><br>
        <strong style="display: block;"><i>dist</i> vs GMD+<i>dist</i></strong>
        <img src="./images/comparing_with_tree_1.png" alt="">
        <br><br>
        <strong style="display: block;">4 mutations, 3 losses, LINEAGE heuristic, variable <i>dist</i></strong>
        <img src="./images/comparison_between_distance_measures_for_4mut_3loss_lineage.png" alt="" srcset="">
        <img src="./images/comparison_of_distance_measures_on_the_parent_heuristic.png" alt="" srcset="">
        <img src="./images/comparison_of_distance_measures_on_the_lineage_heuristic.png" alt="" srcset="">
        <img src="./images/comparison_of_distance_measures_on_the_depth_heuristic.png" alt="" srcset="">
        <br><br>
        <strong style="display: block;">3 losses, Variable gains</strong>
        <img src="./images/3losses_varying_gains_lineage_CASET.png" alt="" srcset="">


    </div>
    <div class="lab-entry">
        <h3 class="date">November 7, 2022</h3>
        <p ondrop="drop(event)" ondragover="allowDrop(event)" class="content">
            Today I am continuing to work on my abstract for the BIBM poster. I also spent a ton of time
            procrastinating by making a <a href="https://quocodile.github.io/lab-notebook/fall.html">site to keep my lab notebook entries</a>
            I also began running some more experiments
            <br><br>
            <strong>Experiment</strong>
            <br><br>
            I ran PARENT+DISC on the space of all trees (probably not the entire space if I'm being honest) with 4 mutations and 4 losses (1 mutation lost 4 times).
            Here's the results...
            <br>
            There were <strong> 138 optimal comparisons out of 240 total comparisons </strong>. With this data, I can create another visualization. The visualization is attached below.
            <br>
            <img src="./images/4mutations_VariableLosses_PARENT_DISC.png" alt="" srcset="">
            <br><br>
            <strong>End experiment</strong>
        </p>
    </div>
    <div class="lab-entry">
        <h3 class="date">November 9, 2022</h3>
        <p ondrop="drop(event)" ondragover="allowDrop(event)" class="content">
            Today I am planning to run some experiments and fleshing out the tree simulation code.
            <br><br>
            <strong>Experiment</strong>
            <br><br>
            Let's do some experiments on the space of 4 mutations, Variable losses. So far, I haven't
            used PARENT-CHILD distance, so let's start with that.
            <br>
            Results for 4 mutations, 2 losses, LINEAGE, PARENT-CHILD: 
            <br>
            Results for 4 mutations, 3 losses, LINEAGE, PARENT-CHILD: <strong>3894/6642</strong> comparisons were optimal
            <br>
            Results for 4 mutations, 4 losses, LINEAGE, PARENT-CHILD: <strong>202/240</strong> comparisons were optimal
            <br><br>
            <strong>End experiment</strong>
        </p>
    </div>
    <div class="lab-entry">
        <h3 class="date">November 12, 2022</h3>
        <p ondrop="drop(event)" ondragover="allowDrop(event)" class="content">
            Finalizing the BIBM abstract.
            <a href="./files/IEEE_BIBM_ISA_Matching_Abstract.pdf">Here's</a> the finalized version.
            I'm also continuing some brainstorming work on simulating trees. 
            <br><br>
            <strong>Experiment</strong>
            <br><br>
            I finished the experiments that I wanted to do on November 9th. Here are some results.
            <br>
            Results for 4 mutations, 2 losses, LINEAGE, PARENT-CHILD: <strong>23723/31152</strong> comparisons were optimal
            <br><br>
            <strong>End experiment</strong>
        </p>
    </div>
    <div class="lab-entry">
        <h3 class="date">November 14, 2022</h3>
        <p ondrop="drop(event)" ondragover="allowDrop(event)" class="content">
            Submitted the BIBM abstract.
        </p>
        <br><br><br><br><br><br><br>
    </div>
    <div class="lab-entry" >
        <h3 class="date">November 15, 2022 -  November 17, 2022</h3>
        <p ondrop="drop(event)" ondragover="allowDrop(event)" class="content">
            Received acceptance for BIBM poster abstract. Worked on simulating trees.
            <br><br>
            Problem: <strong>(SaMT) Simulating All Mutation Trees with n gains, k losses (mutation A lost k times)</strong>
            <br>
            <u>Input</u>: 
                <ul>
                    <li>A mutation tree <i>T</i> with n gains, where each gain is represented by a letter from {A, B, ..., Z}</li>
                    <li>A num k representing k losses (the number of times the mutation A is lost.)</li>
                </ul>
            <br>
            <u>Output</u>: All valid mutation trees with n gains where gain 'A' is lost k times.
            <br><br>
            <u>Algorithm 1</u>:
            <code>
                <br>
                let L be an array.
                <br>
                Algorithm-2(current_tree <i>T'</i>= A mutation tree <i>T</i>, k, list <i>L</i>)
                <br>
                return L
            </code>
            <br>
            <u>Algorithm 2</u>:
            <code>
                <br>
                If k is 0.
                <p style="text-indent: 40px;">Add current_tree to <i>L</i></p>
                Otherwise...
                <br>
                <p style="text-indent: 40px;">
                For each node <i>q</i> in in the current_tree
                <p style="text-indent: 80px;">If the label of <i>q</i> is not 'A-':</p>
                <p style="text-indent: 120px;">If <i>q</i> doesn't have a child that is labeled 'A-':</p>
                </p>
            </code>
            <br><br>
            <strong>Experiment</strong>
            <br><br>
            Continuing to conduct experiment on PARENT-CHILD distance. Now I am combining PARENT(heuristic) with PARENT-CHILD distance.
            <br>
            Here are the results:
            <br>
            Results for 4 mutations, 2 losses, PARENT, PARENT-CHILD: <strong>31152/31152</strong>
            <br>
            Results for 4 mutations, 3 losses, PARENT, PARENT-CHILD: <strong>6642/6642</strong>
            <br>
            Results for 4 mutations, 4 losses, PARENT, PARENT-CHILD: <strong>240/240</strong>
            <br><br>
            I am also combining DEPTH with PARENT-CHILD distance.
            <br>
            Here are the results:
            <br>
            Results for 4 mutations, 2 losses, DEPTH, PARENT-CHILD: <strong>20837/31152</strong>
            <br>
            Results for 4 mutations, 3 losses, DEPTH, PARENT-CHILD: <strong>2184/6642</strong>
            <br>
            Results for 4 mutations, 4 losses, DEPTH, PARENT-CHILD: <strong>34/240</strong>
            <br><br>
            <strong>End experiment</strong>
        </p>
    </div>
    <div id="math-toolbar">
        <div draggable="true" ondragstart="drag(event)" class="math-icon">&plus;</div>
        <div draggable="true" ondragstart="drag(event)" class="math-icon">&minus;</div>
        <div draggable="true" ondragstart="drag(event)" class="math-icon">&times;</div>
        <div draggable="true" ondragstart="drag(event)" class="math-icon">&divide;</div>
    </div>
    <script src="fall.js"></script>
</body>
</html>
